### PR TITLE
Spring Boot Tests

### PR DESCRIPTION
**Pull request by deikunx**

### CODE REVIEW

**Comment by deikunx** [comment]
image

File: src/test/java/com/bookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        expected.add(new BookDto().setTitle("Book 3")
+                .setPrice(BigDecimal.valueOf(14.99))
+                .setAuthor("Author 3")
+                .setIsbn("ISBN456789123")
+                .setDescription("Description 3")
+                .setCoverImage("image3.jpg")
+                .setCategoryIds(Collections.emptySet())
+                .setId(3L));
+
+        when(bookService.findAll(any(Pageable.class))).thenReturn(expected);
```
Comment (aaananas) [inline src/test/java/com/bookstore/controller/BookControllerTest.java:93]: controller tests should be integrational))

File: src/test/java/com/bookstore/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                .setDescription(requestDto.getDescription())
+                .setCoverImage(requestDto.getCoverImage())
+                .setAuthor(requestDto.getAuthor())
+                .setCategoryIds(requestDto.getCategoryIds());
+
+        when(bookService.save(eq(requestDto))).thenReturn(response);
+
+        mockMvc.perform(post("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(new ObjectMapper().writeValueAsString(requestDto)))
```
Comment (aaananas) [inline src/test/java/com/bookstore/controller/BookControllerTest.java:129]:
```suggestion
                        .content(objectMapper.writeValueAsString(requestDto)))
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Comment by deikunx** [comment]
image

**Pull request merged**